### ISSUE TITLE
Chef run timeout on windows

### ISSUE THREAD

**Issue by Annih** [chef/chef#2985]
Hi,

I'm facing a small issue with some chef run that take a lot of time to complete.
[Before Chef 12.0.0](https://github.com/chef/chef/commit/a463b6df7ed3003447644ca99ea52f7ef0b404ea) it's not a big deal because MixlibShellout 1.4 wasn't terminating the "timeouted" process. But since Chef is now using the [latest shellout version](https://github.com/chef/mixlib-shellout/pull/79), it'll sadly kill the running task.
We should either disable the timeout feature for the chef_run, or add a way to tune it.

### Version:
all including latest (12.1.0.rc.0)

### Environment:
Windows

### Scenario:
My chef run is performing a synchronous update of WSUS catalog, which can take a lot of time in certain condition.

### Steps to Reproduce:
You just need a chef run longer than 10minutes, so a recipe with a sleep 600.

### Expected Result:
Chef run complete cleanly

### Actual Result:
* On Chef <12.0.0 chef run complete properly, but reports timeout (not clean)
```
[2015-02-26T00:00:00+00:00] ERROR: command timed out:
---- Begin output of chef-client  --no-fork -c C:/chef/client.rb -L C:/chef/client.log ----
STDOUT:
STDERR:
---- End output of chef-client  --no-fork -c C:/chef/client.rb -L C:/chef/client.log ----
```
* Chef >=12.0.0, the process might be killed why performing a long running task. *not tested*

**Comment by jdmundrawala** [comment]
Are you running chef client as a Windows service?

**Comment by Annih** [comment]
Yes I'm running chef client as a Windows service :)
Your PR seems correct for me

**Comment by jdmundrawala** [comment]
Merged in, will be released in 12.1.0
